### ISSUE TITLE
Re-saving a DynamicDocument with an EmbeddedDocument field crashes mongoengine.

### ISSUE THREAD

**Issue by jvangael** [hmarr/mongoengine#561]
Creating a DynamicDocument with an embedded document field works fine. However, when opening the document again, the embedded document has its _dynamic field set to True. When resaving the dynamic document, mongoengine will crash because the embedded document's _dynamic attribute implies that dynamic fields should be available, which is not the case.

**Comment by rozza** [comment]
This test case works for me:

```
def test_dynamic_and_embedded(self):

    class Address(EmbeddedDocument):
        city = StringField()

    class Person(DynamicDocument):
        name = StringField()
        address = EmbeddedDocumentField(Address)
        meta = {'allow_inheritance': True}

    Person.drop_collection()

    Person(name="Ross", address=Address(city="London")).save()

    person = Person.objects.first()
    person.address.city = "Lundenne"
    person.save()

    self.assertEqual(Person.objects.first().address.city, "Lundenne")

    person = Person.objects.first()
    person.address = Address(city="Londinium")
    person.save()

    self.assertEqual(Person.objects.first().address.city, "Londinium")
```

Using master and v0.6.20

**Comment by rozza** [comment]
Fixed in MongoEngine/mongoengine@529c5225944943ca541ff5bba570535af5363605

**Comment by rozza** [comment]
Just an FYI - we're moving!  Please report issues on: http://github.com/MongoEngine/mongoengine
